Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver_Tests.swift

```diff
@@ -0,0 +1,336 @@
+            message.isSentByCurrentUser
+        }
+    }
+
+    func test_bubble() {
+        // Iterate channel history and check `.bubble` is always there
+        assertOptionIsIncluded(.bubble) { _, _ in
+            true
+        }
+    }
+
+    func test_continuousBubble() {
+        // Iterate channel history and check `.continuousBubble` is there if conditions are met
+        assertOptionIsIncluded(.continuousBubble) { message, isStandaloneOrLastInSequence in
+            !isStandaloneOrLastInSequence || message.isRootOfThread || message.isPartOfThread
+        }
+    }
+
+    func test_avatarSizePadding() {
+        // Iterate channel history and check `.avatarSizePadding` is there if conditions are met
+        assertOptionIsIncluded(.avatarSizePadding) { message, isStandaloneOrLastInSequence in
+            !isStandaloneOrLastInSequence && !message.isSentByCurrentUser
+        }
+    }
+
+    func test_metadata() {
+        // Iterate channel history and check `.metadata` is there if conditions are met
+        assertOptionIsIncluded(.metadata) { _, isStandaloneOrLastInSequence in
+            isStandaloneOrLastInSequence
+        }
+    }
+
+    func test_onlyVisibleForCurrentUser() {
+        // Iterate channel history and check `.onlyVisibleForYouIndicator` is there if conditions are met
+        assertOptionIsIncluded(.onlyVisibleForYouIndicator) { message, _ in
+            message.onlyVisibleForCurrentUser
+        }
+    }
+
+    func test_text() {
+        // Iterate channel history and check `.text` is there if conditions are met
+        assertOptionIsIncluded(.text) { message, _ in
+            message.textContent?.isEmpty == false
+        }
+    }
+
+    func test_avatar() {
+        // Iterate channel history and check `.avatar` is there if conditions are met
+        assertOptionIsIncluded(.avatar) { message, isStandaloneOrLastInSequence in
+            isStandaloneOrLastInSequence
+                && !message.isSentByCurrentUser
+                && message.deletedAt == nil
+        }
+    }
+
+    func test_authorName() {
+        let channels: [ChatChannel] = [
+            .mock(cid: .unique),
+            .mockDMChannel()
+        ]
+
+        for channel in channels {
+            // Iterate channel history and check `.authorName` is there if conditions are met
+            assertOptionIsIncluded(.authorName, channel: channel) { message, isStandaloneOrLastInSequence in
+                isStandaloneOrLastInSequence
+                    && !message.isSentByCurrentUser
+                    && message.deletedAt == nil
+                    && !channel.isDirectMessageChannel
+            }
+        }
+    }
+
+    func test_quotedMessage() {
+        // Iterate channel history and check `.quotedMessage` is there if conditions are met
+        assertOptionIsIncluded(.quotedMessage) { message, _ in
+            message.quotedMessage != nil && message.deletedAt == nil
+        }
```

Comment:
I don't think this is the best way to test this. Tests should be as dummy as possible and should not contain any logic, and with this, we are not really testing that "quotedMessage" added the `.quotedMessage` option. For example, if there was a bug, and `isSentByCurrentUser` option

Reply:
Yeah, I see what you mean. Makes sense 👍 Let me have one more iteration on this

Reply:
✅ Updated